internal/plugin/extract.go

```diff
@@ -0,0 +1,83 @@
+package plugin
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+const pluginsExtractsFolder = pluginsFolder + "/extracts"
+
+func init() {
+	_ = os.MkdirAll(pluginsExtractsFolder, 0755)
+}
+
+func extractPlugin(filePath string) (*string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file for extraction: %v", err)
+	}
+	defer file.Close()
+
+	var reader io.Reader = file
+	// TODO: there's probably a better way of doing this without relying on the file extension
+	if strings.HasSuffix(filePath, ".gz") {
```

Comment:
Grabbing the first few bytes of the file and doing:

```
if bytes.HasPrefix(buf, []byte{0x1F, 0x8B, 0x08}) {
```

can be a nicer way to handle this.